lib/sql-string.js

```diff
@@ -35,6 +35,11 @@ function escape(val, timeZone, dialect, format) {
     break;
   }
 
+  // moment
+  if (val.toDate) {
```

Comment:
Passing anything to moment can lead to some crazy issues. Like passing `really not a date` to `moment`. Rather than trying to support every possible string out there, we should stick with well defined and valid moment object. Since @felixfbecker has raised this point. I think adding `isValid` with `isMoment` check will not hurt either.

@felixfbecker I am not against the duck typing :), But for this particular case I would prefer valid moment object.